### PR TITLE
add new custom report

### LINKED ISSUE #1201: Custom regular report for Stanford Medicine Center for Improvement (SMCI)
Consider producing a custom report on a regular basis that includes both publications for users currently in profiles, as well as users not currently in profiles.

Most likely it will be a once a month report.  They need to have a citation or be able to build the citation they want.  Therefore, we were going to provide them with the most recent approved publications (from the last date received) with the following:

- First Published Date
- DOI URL
- Chicago Citation Format
- APA Citation Format
- MLA Citation Format

Additionally, we wanted to provide them with the separated fields that form the citation in the case they want to do their own formatting for the citation display on their website.  So that would be, for example:

- Title
- Article No.
- Pages
- DOI / DOI URL
- Journal Title
- Publisher
- Year Published
- Volume
- Issue
- Supplement Number
- Authors - First name, Middle Initial, Last Name

This would also need to include the author's details, when available:

- First Name
- Last Name
- Position
- sunet id
- profile id
- univ id
- email address

The thought is that possibly it would make sense, if you can pull the information for those without profiles, that you produce the full report for everyone that also includes the data for those with profiles (thus some of the fields noted above), so that the report is coming from one source and combined.  The number of researchers is from 94 - 130 from the current information we have.  Out of these there were around 33 that did not have profiles or active profiles and there a number that do not have full profiles that have the publication import option.  Also, we have staff members in the list who would also not have the publication import turned on by default, as this is only on by default for faculty and postdocs.

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

Fixes #1201 

A specific custom report that combines publications from our database with WoS name queries for people not in our database.

## How was this change tested?

Running the report on my laptop with various iterations of data.  New unit test to verify basic runnability.

## Which documentation and/or configurations were updated?

None yet, but will add to the wiki.

### CODE REVIEW

File: lib/smci_report.rb [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+#   specified in https://github.com/sul-dlss/sul_pub/issues/1201
+
+# October 2020
+
+# You must have an input csv file with the following columns (including a header row):
+#   sunetid,cap_profile_id,first_name,middle_name,last_name,institutions
+#
+#  For authors that exist in profiles, you can specify either a sunet or cap_profile_id (sunet preferred if both specified)
+#  For authors that do NOT exist in profiles, you MUST leave the cap_profile_id and sunet field columns blank, and then you must specify first name and last name
+#    and optionally middle name a comma delimtied list of institutions (defaults to 'stanford' if no institutions provided)
```
Comment (justinlittman) [inline lib/smci_report.rb:13]: typo = delimtied

File: lib/smci_report.rb [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+# Run with:
+#
+# smci = SMCIReport.new(input_file: 'tmp/input.csv', output_file: 'tmp/output_file.csv', date_since: '1/1/2000', time_span: '1year')
```
Comment (justinlittman) [inline lib/smci_report.rb:17]: What is the context for executing these commands?

Comment (justinlittman) [reply lib/smci_report.rb:17]: Also, mention that this is available as a rake task.

**Review by justinlittman — COMMENTED** [review]
This seems like a reasonable, minimal effort approach.

Add documentation to README?

File: lib/smci_report.rb (hunk 1 above)
Comment (peetucket) [reply lib/smci_report.rb:13]: thanks, will fix

File: lib/smci_report.rb (hunk 2 above)
Comment (peetucket) [reply lib/smci_report.rb:17]: right, this was before there was a rake task...but I will indicate you can run this from the Rails console as well if you want

**Comment by peetucket** [comment]
> This seems like a reasonable, minimal effort approach.
> 
> Add documentation to README?

Will add to the wiki, which is the place where this sort of custom scripty stuff is documented for this project.

File: lib/smci_report.rb [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    mesh = pub_hash[:mesh_headings] ? pub_hash[:mesh_headings].map { |h| h[:descriptor][0][:name] }.compact.reject(&:empty?).join('; ') : ''
+    if author # if this is a profiles author, then we will output this info as well
+      author_last_name = author.last_name
+      author_first_name = author.first_name
+      author_sunet = author.sunetid
+      author_cap_profile_id = author.cap_profile_id
+      author_univ_id = author.university_id
+      author_email = author.email
+    else
+      author_last_name = author_first_name = author_sunet = author_cap_profile_id = author_univ_id = author_email = ""
```
Comment (jcoyne) [inline lib/smci_report.rb:187]: Can you add a test for this branch?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: lib/smci_report.rb (hunk 3 above)
Comment (peetucket) [reply lib/smci_report.rb:187]: added a few more tests to hit this branch of the code

**Review by jcoyne — APPROVED** [review]

**Pull request merged**